### ISSUE TITLE
[BUG][Grid] Forced GridColumnMenuWrapper scroll on focus

### ISSUE THREAD

**Issue by CoryMut** [telerik/kendo-react#1733]
## I'm submitting a...

* Bug report 

## Current behavior

Clicking on the ColumnMenu within the grid causes the column to randomly move to the center of the grid. This is confusing even when expected. The initial click also does not open the ColumnMenu, forcing the user to locate the newly moved column in order to press it again.

## Expected behavior

Clicking the ColumnMenu should only open the menu. If there is some desire to scroll, make it a prop.
## Minimal reproduction of the problem with instructions

1) Open CodeSandbox link [here](https://codesandbox.io/s/falling-cloud-xhv6q9).
2) Make the browser section sufficiently large so there are several columns and the grid has horizontal scroll.
3) Click on any of the ColumnMenu buttons for any column on the right hand side.
## What is the motivation or use case for changing the behavior?

The user experience is significantly worse since this change. I believe the change occurred in v5.19.0.
## Environment

Package versions:

```
{
    "@progress/kendo-data-query": "latest",
    "@progress/kendo-drawing": "latest",
    "@progress/kendo-licensing": "latest",
    "@progress/kendo-react-animation": "latest",
    "@progress/kendo-react-buttons": "latest",
    "@progress/kendo-react-data-tools": "latest",
    "@progress/kendo-react-dateinputs": "latest",
    "@progress/kendo-react-dropdowns": "latest",
    "@progress/kendo-react-form": "latest",
    "@progress/kendo-react-grid": "latest",
    "@progress/kendo-react-inputs": "latest",
    "@progress/kendo-react-intl": "latest",
    "@progress/kendo-react-popup": "latest",
    "@progress/kendo-react-progressbars": "latest",
    "@progress/kendo-react-treeview": "latest",
    "@progress/kendo-svg-icons": "latest",
    "react": "^18.0.0",
    "react-dom": "^18.0.0"
  }
```

Browser:

- Chrome (desktop) Version 118.0.5993.89 (Official Build) (64-bit)

System:
- Node version: v18.18.1 
- NPM version: 10.2.0
- Platform:  Windows

**Comment by vveesseelliinnaa** [comment]
Th reported issue is a duplicate of the following one:
https://github.com/telerik/kendo-react/issues/1233

**Comment by CoryMut** [comment]
I disagree with closing and labeling this issue as a duplicate. #1233 is referring to vertical scrolling of the page. This ticket is referring to a deliberate decision in v5.19.0 to scroll the grid horizontally when the ColumnMenu is clicked. It is explicitly mentioned in the [changelog](https://www.telerik.com/kendo-react-ui/components/changelogs/ui-for-react/). Using a version of the grid prior to v5.19.0 works as expected.
![Screenshot 2023-10-23 074311](https://github.com/telerik/kendo-react/assets/59666828/e3fcb607-6ae9-4142-893d-3e3f797b4a0a)

**Comment by vveesseelliinnaa** [comment]
Thank you for the follow-up, @CoryMut! 

We reviewed both of the reported items again and it turned out that the second commented scenario from the linked issue is actually related to this report. I will move this report here and update the severity of this item to "critical".

video of the issue : http://somup.com/c061bBBtFP

Reported also in Ticket ID: 1625978

**Comment by iadnonov** [comment]
Fixed in our dev channel version [v6.0.3-dev.202310260856](https://www.telerik.com/kendo-react-ui-develop/components/changelogs/ui-for-react/#v6.0.3-dev.202310260856)
